tools/reactive-controllers/src/FocusGroup.ts

```diff
@@ -266,6 +265,28 @@ export class FocusGroupController<T extends HTMLElement>
         });
         this.prevIndex = this.currentIndex;
         this.currentIndex = targetIndex > -1 ? targetIndex : this.currentIndex;
+
+        if (this.isRelatedTargetOrContainAnElement(event)) {
+            this.hostContainsFocus();
+        }
+    };
+
+    handleClick = (): void => {
+        // Manually set the tabindex to 0 for the current element on receiving focus (from keyboard or mouse)
+        const elements = this.elements;
+        if (!elements.length) return;
+        let focusElement = elements[this.currentIndex];
+        if (this.currentIndex < 0) {
+            return;
+        }
+        if (!focusElement || !this.isFocusableElement(focusElement)) {
+            this.setCurrentIndexCircularly(1);
+            focusElement = elements[this.currentIndex];
+        }
+        if (focusElement && this.isFocusableElement(focusElement)) {
```

Comment:
It would be great if you can add some documentation block on this method!

Reply:
Done